### ISSUE TITLE
Formatting error on Windows machines

### ISSUE THREAD

**Issue by Eprince-hub** [Sec-ant/prettier-plugin-embed#84]
The `prettier-plugin-embed-embed` does not load the `prettier-plugin-sql` on Windows machine and shows this error on the VS Code devTools when saving a file. 

```bash
Error: Cannot format embedded language identified by "sql", because plugin "prettier-plugin-sql" is not loaded.
```
![image (3)](https://github.com/Sec-ant/prettier-plugin-embed/assets/74430629/b14df508-40c3-4713-bfc8-3626f424d8c5)

The error does not show in the output panel of VS Code

This the content of the prettier config file
```ts
/** @type {import('prettier').Config} */
const prettierConfig = {
  plugins: ['prettier-plugin-embed', 'prettier-plugin-sql'],
  singleQuote: true,
  trailingComma: 'all',
};

/** @type {import('prettier-plugin-embed').PrettierPluginEmbedOptions} */
const prettierPluginEmbedConfig = {
  embeddedSqlComments: ['sql'],
  embeddedSqlTags: ['sql'],
};

/** @type {import('prettier-plugin-sql').SqlBaseOptions} */
const prettierPluginSqlConfig = {
  language: 'postgresql',
  keywordCase: 'upper',
  identifierCase: 'lower',
  dataTypeCase: 'lower',
  functionCase: 'lower',
  expressionWidth: 30,
};

const config = {
  ...prettierConfig,
  ...prettierPluginEmbedConfig,
  ...prettierPluginSqlConfig,
};

export default config;
```

The SQL code that should be formatted
```ts
export async function up(sql: Sql) {
await sql`
CREATE TABLE workshops (
id integer PRIMARY key generated always AS identity,
title varchar(70) NOT NULL,
workshop_date varchar(40) NOT NULL,
timeframe varchar(40) NOT NULL,
location varchar(40),
category varchar(40),
image varchar(50) NOT NULL,
price integer NOT NULL,
description varchar NOT NULL
    )
  `;
}
```

I am using the following versions for the plugins
```bash
"prettier": "^3.2.5",
"prettier-plugin-embed": "^0.4.13",
"prettier-plugin-sql": "^0.18.0",
```

**Comment by Sec-ant** [comment]
Hi Victor,

Thank you very much for this detailed report. It's very helpful. I pushed a fix in #85. Will you check if the test package in https://github.com/Sec-ant/prettier-plugin-embed/pull/85#issuecomment-1953422584 does resolve this issue?

**Comment by Eprince-hub** [comment]
Hello, @Sec-ant,

Thank you so much for the prompt answer and fix. I can confirm that the fix pushed in https://github.com/Sec-ant/prettier-plugin-embed/pull/85 did fix the issue, and now the SQL codes can be formatted on save, but I want to know if there is any downside in this approach (Removing the throwIfPluginIsNotFound check). What was the purpose of this code and why is it no longer necessary?

https://github.com/Sec-ant/prettier-plugin-embed/assets/74430629/678483c6-b48c-4cb3-9d97-c362f69f018d

**Comment by Sec-ant** [comment]
> What was the purpose of this code

The function `throwIfPluginIsNotFound` is supposed to print an explaining message when the formatting of embedded languages failed silently because of the user not loading the corresponding language plugins (like `prettier-plugin-sql` or plugins for other languages).

> Why is it no longer necessary

TL;DR, this function does more harm than good.

The paths of the loaded plugins are registered in the `options` object in `Prettier` which can be accessed by this plugin. So this function finds the paths of all the loaded plugins and checks if the required plugin is loaded. If it cannot find the required plugin it then throws and exits early without trying to format the embedded blocks, and prints the explaining message.

However there're several issues with this approach:

1. A user can fork a language plugin and rename or relocate it for whatever reason without modifying its function. And with this approach the user cannot use the forked plugin because the check fails.
2. To pair the plugin path with the required plugin name requires some additional work to deal with path separators on different platforms (Win vs Linux vs MacOS) or runtimes (node vs deno vs web etc.). Additionally the function more or less works in a heuristic way so it is error-prone.
3. The initial intention to introduce this function is to provide explaining messages to the users but due to the logging service used in the VSCode extension is unavailable to the plugin, it can only log messages to the DevTools console which basically defeat the purpose.

So I decide to remove this function.

**Comment by Sec-ant** [comment]
You're welcome to share them if you have better ideas @Eprince-hub, otherwise I'm going to move forward and release a new version including this fix.

**Comment by Eprince-hub** [comment]
@Sec-ant, amazing! I think it makes sense to go on with this fix, as you have explained above 👍
Thank you

**Comment by Sec-ant** [comment]
The fix is shipped in [v0.4.14](https://github.com/Sec-ant/prettier-plugin-embed/releases/tag/v0.4.14).
